sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp

```diff
@@ -1100,7 +1119,7 @@ namespace Azure { namespace Storage { namespace Blobs {
         unused(body);
         auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Get, url);
         request.AddHeader("Content-Length", "0");
```

Comment:
If the request/response header keys are all lower case, please be consistent for setting/parsing the response.

Reply:
In addition, in the [RFC](https://tools.ietf.org/html/rfc2616) it's defined `Content-Length` rather than `content-length`. So what's wrong with writing it correctly?

Reply:
Then why we are parsing as `content-length`? I don't think we are writing the correct code when we are sending or parsing with different header keys.
In an ideal world, `"Content-Length"` will be defined as a header key string constant, and it is intuitive to set/get using this constant. It is difficult for anyone who maintains this library to remember that set/get needs to use different header keys for the same headers for requests and responses.

Reply:
These are two different things. Let's focus on the first one. Do you think it's reasonable at all to tell an HTTP client that we want to set header `Content-Length`?

Reply:
> 4.2 Message Headers
> 
>    HTTP header fields, which include general-header (section 4.5),
>    request-header (section 5.3), response-header (section 6.2), and
>    entity-header (section 7.1) fields, follow the same generic format as
>    that given in Section 3.1 of RFC 822 [9]. Each header field consists
>    of a name followed by a colon (":") and the field value. **Field names
>    are case-insensitive.** The field value MAY be preceded by any amount
>    of LWS, though a single SP is preferred. Header fields can be
>    extended over multiple lines by preceding each extra line with at
>    least one SP or HT. Applications ought to follow "common form", where
>    one is known or indicated, when generating HTTP constructs, since
>    there might exist some implementations that fail to accept anything
RFC also says that `Field names
are case-insensitive.` So `Content-Length` and `content-length` even `CONTent-LenGtH` should work at all time.
I think what I want to have is case-insensitive compares when getting the response header value, then there won't be any problems anymore.

Reply:
Since it's case-insensitive, I can set header by `Content-Length` and get header by `content-length`.

Reply:
To some extent, i agree with you on the map with case-insensitive comparator idea. Anyway, it's not priority.